### PR TITLE
Flake-build-image

### PR DESCRIPTION
**Pull request by Ichmed**
Only supports podman so far

### CODE REVIEW

File: flake-ctl/flake-ctl-build-rpmbuild/src/main.rs [hunk 1]
```diff
@@ -144,31 +149,34 @@ impl RPMBuilder {
...
 
         // TODO: maybe use faster templating here
         // TODO: Drop line with empty information completely
         let data = self.template.join(config.engine().pilot());
         let requires = fs::read_to_string(&data).context(format!("Failed to load pilot specific data, {data:?}"))?;
+        
+        let mut symlinks: Vec<String> = vec![];
+        if let Some((first, rest)) = config.runtime().get_symlinks() {
+            let first = first.to_string_lossy();
+            symlinks.push(format!("ln -s %{{_bindir}}/%{{_flake_pilot}}-pilot {first}"));
```
Comment (isbm) [inline flake-ctl/flake-ctl-build-rpmbuild/src/main.rs:164]: What's wrong with `fs::symlink` ?

File: flake-ctl/flake-ctl-build-rpmbuild/src/main.rs [hunk 2]
```diff
@@ -144,31 +149,34 @@ impl RPMBuilder {
...
+            symlinks.extend(rest.map(|path| format!("ln {first} {}", path.to_string_lossy())))
```
Comment (isbm) [inline flake-ctl/flake-ctl-build-rpmbuild/src/main.rs:165]: That one is not symbolic, BTW. But still why external syscall?

File: flake-ctl/flake-ctl-build/src/builder.rs [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+                    std::process::Command::new("flake-ctl")
+                        .arg(pilot)
+                        .arg("remove")
+                        .arg("--root")
+                        .arg(root)
+                        .arg("--app")
+                        .arg(app)
+                        .status()?;
+                }
+                // remove_file(path)?;
```
Comment (isbm) [inline flake-ctl/flake-ctl-build/src/builder.rs:201]: ?

File: flake-ctl/flake-ctl-build-rpmbuild/src/main.rs (hunk 1 above)
Comment (Ichmed) [reply flake-ctl/flake-ctl-build-rpmbuild/src/main.rs:164]: the fact that this is going into a script file and not being executed on the machine running the builder ^^

File: flake-ctl/flake-ctl-build/src/builder.rs (hunk 3 above)
Comment (Ichmed) [reply flake-ctl/flake-ctl-build/src/builder.rs:201]: already changed

**Review by isbm — APPROVED** [review]

**Pull request merged**